### PR TITLE
[doc] Add reference security/use_cases

### PR DESCRIPTION
**Pull request by moidx**
This commit adds reference use cases for:

* Platform Integrity Module: Titan datacenter use case.
* Security Key: FIDO2 on USB 1.1 use case.

Closes issue lowrisc/opentitan#1119

Signed-off-by: Miguel Osorio <miguelosorio@google.com>

### CODE REVIEW

File: doc/security/use_cases/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+title: "OpenTitan Use Cases"
+---
+
+
+## Overview
+
+OpenTitan’s mission is to raise the security bar industry wide by implementing a transparent, logically secure hardware root of trust with wide application.
```
Comment (imphil) [inline doc/security/use_cases/index.md]: nit: please use an ASCII `'`

File: doc/security/use_cases/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+This document describes some of those use cases for OpenTitan.
+These range from data center integrations to implement platform integrity solutions, to embedded security applications such as security keys and smart cards.
```
Comment (imphil) [inline doc/security/use_cases/index.md]: Is "data center integrations to implement platform integrity solutions" one thing? This sentence could probably be split up into two to make it less dense and easier to understand.

File: doc/security/use_cases/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+References to relevant specifications and certification targets are noted where use cases are backed by industry standards.
+
+
+## Platform Integrity Module
+
+An OpenTitan IC used as a Platform Integrity Module interposes between a platform’s boot flash and its main boot devices such as BMC, PCH and CPU.
```
Comment (imphil) [inline doc/security/use_cases/index.md]: * nit: ASCII `'`
* Can you spell out the BMC and PCH acronyms?

File: doc/security/use_cases/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+* Owner Cryptographic Identity: The Silicon Owner is required to generate an identity as part of the Ownership transfer flow.
+  Owner identities are bound to the Hardware Identity and the software version running on the device.
+  Owner identities are used in Silicon Ownership attestation flows and as a root component of Application keys.
+* Application Keys: Keys bound to the owner identity and the application version running on the device.
+  Application keys are provisioned in most cases when the application runs for the first time.
+  The purpose of each key is configured at the application layer and enforced by the kernel.
+
+### Performance Requirements
+
+Performance requirements are derived from integration specific requirements.
```
Comment (imphil) [inline doc/security/use_cases/index.md]:
```suggestion
Performance requirements are derived from integration-specific requirements.
```

File: doc/security/use_cases/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+* Boot Time: Maximum time allowed to get to device kernel serving state from cold reset.
+* Resume Time: Maximum time allowed to get to device kernel serving state from sleep.
+* External Flash Verification Time: Maximum time allowed for verification of external flash as part of platform boot verified boot implementation.
+  Defined in milliseconds for a given flash partition size.
+
+### Packaging Constraints
+
+* Non-HDI packaging is required.
+* (Proposed) Device packaging QR code with device id linkable to manufacturing data.
```
Comment (imphil) [inline doc/security/use_cases/index.md]: id -> ID

File: doc/security/use_cases/index.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+### Additional Requirements
+
+#### Memory Requirements
+
+* At least 512KB of flash storage with 2 partitions, 4KB page size, 100K endurance cycles.
+  1MB flash would be ideal to allow for future code size growth assuming the CPU supports compressed instructions.
```
Comment (imphil) [inline doc/security/use_cases/index.md]: "assuming the CPU supports compressed instructions" sounds a bit odd - Ibex supports compressed instructions.

**Comment by tjaychen** [comment]
@moidx 
separate but related..all the security docs are orphaned in opentitan docs right now.  Do you mind linking them up?

**Comment by moidx** [comment]
> @moidx
> separate but related..all the security docs are orphaned in opentitan docs right now. Do you mind linking them up?

@tjaychen I am working on getting an index.md doc cleared for review. Once I have that, I will link the security docs there, and use the index.md to link to the top doc level. ETA EOW or early next week the latest.

**Comment by tjaychen** [comment]
thank you!

File: doc/security/use_cases/index.md (hunk 2 above)
Comment (tjaychen) [reply doc/security/use_cases/index.md]: agree with @imphil .  I feel like data center integrity solutions is probably sufficient.

File: doc/security/use_cases/index.md (hunk 1 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done

File: doc/security/use_cases/index.md (hunk 2 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done.

File: doc/security/use_cases/index.md (hunk 3 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done

File: doc/security/use_cases/index.md (hunk 4 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done

File: doc/security/use_cases/index.md (hunk 5 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done

File: doc/security/use_cases/index.md (hunk 6 above)
Comment (moidx) [reply doc/security/use_cases/index.md]: Done. Removed compressed instructions comment.

**Comment by sjgitty** [comment]
I'm late to the party here, scanning now. Initial thought is the same one @tjaychen 
brought up in the lack of referencing index.md file.

File: doc/security/use_cases/index.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+  * AES: AES-CTR NIST 800-38A. Used to wrap keys and encrypt data stored in internal flash.
+
+
+### Provisioning Requirements
+
+Provisioning an OpenTitan device is performed in two steps:
+
+* Device Personalization: The device is initialized with a unique cryptographic identity endorsed by a Transit PKI which is only used to support initial Ownership Transfer.
+* Ownership Transfer: Ownership is assigned to a user that has the ability to run software on the device.
+  As Silicon Owner, the user can generate a cryptographic identity strongly associated to the hardware and the software version running on the device.
```
Comment (sjgitty) [inline doc/security/use_cases/index.md]: Do we want to reference the definitions in `doc/security/logical_security_model`?

File: doc/security/use_cases/index.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+### Packaging Constraints
+
+* Non-HDI packaging is required.
+* (Proposed) Device packaging QR code with device ID linkable to manufacturing data.
+
+### Additional Requirements
+
+#### Memory Requirements
+
+* At least 512KB of flash storage with 2 partitions, 4KB page size, 100K endurance cycles.
```
Comment (sjgitty) [inline doc/security/use_cases/index.md]: not trying to throw a bomb, but we might want to agree among ourselves on the great kB vs KB (vs KiB) debate.

Comment (sjgitty) [reply doc/security/use_cases/index.md]: Or even more fun bomb: personalization vs personalisation :)

Comment (sjgitty) [reply doc/security/use_cases/index.md]: the bomb would say: 51 uses of kB in our repo, 21 of KB

File: doc/security/use_cases/index.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+The following list of peripheral requirements is speculative at the moment and subject to change based on platform integration requirements:
+
+* SPI Host/Device:
+  * Dual support. Quad support needs to be evaluated.
+  * Required features for EEPROM mode:
+      - Passthrough boot flash interface with support for EEPROM command handling/filtering.
+      - Access to on-die ram and flash memory regions.
+      - Mailbox interface with support for custom opcode commands.
+* eSPI: Enhanced Serial Peripheral (eSPI) which enables asynchronous processing of boot flash commands issued by a boot device supporting the eSPI protocol and serviced by OpenTitan.
```
Comment (sjgitty) [inline doc/security/use_cases/index.md]: eSPI truly a requirement, or "needs to be evaluated"?

**Pull request merged**